File: YLPageScrollViewAppreance.m
//
//  YLPageScrollViewAppreance.m
//  YLPageScrollView
//
//  Created by Youngly on 2017/3/13.
//  Copyright © 2017年 YL. All rights reserved.
//

#import "YLPageScrollViewAppreance.h"
#import "YLPageScrollViewConfigure.h"

@implementation YLPageScrollViewAppreance
- (instancetype)init
{
    self = [super init];
    if (self) {
        [self initialize];
    }
    return self;
}

- (void)initialize {
    
    self.titleFont = [UIFont systemFontOfSize:14];
    self.titleViewHeight = 40.0;
    self.titleOffset = 8;
    self.titleMargin = 20;
    self.botomLineHeight = 3;
    self.maxScaleRatio = 1.15;
    
    self.titleViewColor = [UIColor whiteColor];
    self.titleNormalColor = [UIColor orangeColor];
    self.titleSelectedColor = [UIColor purpleColor];
    self.bottomLineColor = self.titleSelectedColor;
    
    self.isScrollEnable = YES;
    self.isShowBottomLine = YES;
    self.isNeedScale = NO;
}
- (NSArray *)normalColors {
    if (!_normalColors) {
        CGFloat normalRed = 0;
        CGFloat normalGreen = 0;
        CGFloat normalBlue = 0;
        [self.titleNormalColor getRed:&normalRed green:&normalGreen blue:&normalBlue alpha:nil];
        _normalColors = @[@(ABS(normalRed)), @(ABS(normalGreen)), @(ABS(normalBlue))];
    }
    return _normalColors;
}
- (NSArray *)selectedColors {
    if (!_selectedColors) {
        CGFloat normalRed = 0;
        CGFloat normalGreen = 0;
        CGFloat normalBlue = 0;
        [self.titleSelectedColor getRed:&normalRed green:&normalGreen blue:&normalBlue alpha:nil];
        _selectedColors = @[@(ABS(normalRed)), @(ABS(normalGreen)), @(ABS(normalBlue))];
    }
    return _selectedColors;
}

- (NSArray *)deltaColors {
    if (!_deltaColors) {
        _deltaColors = @[@(ABS([self.normalColors[0] doubleValue]-[self.selectedColors[0] doubleValue])), @(ABS([self.normalColors[1] doubleValue]-[self.selectedColors[1] doubleValue])), @(ABS([self.normalColors[2] doubleValue]-[self.selectedColors[2] doubleValue]))];
    }
    return _deltaColors;
}
@end

File: YLPageScrollView/YLPageScrollView/YLPageScrollViewAppreance.swift
//
//  YLPageScrollViewAppreance.swift
//  YLPageScrollView
//
//  Created by Youngly on 2017/3/12.
//  Copyright © 2017年 YL. All rights reserved.
//

import UIKit

class YLPageScrollViewAppreance {

    /// 标题文字大小
    var titleFont : UIFont = UIFont.systemFont(ofSize: 14)
    /// 标题滚动视图的高度
    var titleViewHeight : CGFloat = 40.0
    /// 标题左右距离屏幕的偏移量
    var titleOffset : CGFloat = 8
    /// 标题文字之间的间距
    var titleMargin : CGFloat = 20
    /// 底部线的高度
    var botomLineHeight : CGFloat = 3.0
    /// 文字缩放系数
    var maxScaleRatio : CGFloat = 1.2
    
    /// 标题文字普通颜色
    var titleNormalColor : UIColor = UIColor(r: 200, g: 200, b: 200)
    /// 标题文字选中颜色
    var titleSelectedColor : UIColor = UIColor(r: 50, g: 50, b: 50)
    /// 标题底部线的颜色, 默认=titleNormalColor
    lazy var bottomLineColor : UIColor = {
        let bottomLineColor = self.titleNormalColor
        return bottomLineColor
    }()
    
    /// 是否允许标题栏滚动
    var isScrollEnable : Bool = false
    /// 是否显示底部线
    var isShowBottomLine = false
    /// 是否需要缩放标题
    var isNeedScale = true
}
